File: Loenn/entities/ELS_Termina_Final_Boss.lua
local utils = require("utils")

local elsTerminaFinalBoss = {}

elsTerminaFinalBoss.name = "MaggyHelper/ELSTerminaFinalBoss"
elsTerminaFinalBoss.depth = -12500
elsTerminaFinalBoss.placements = {
    name = "default",
    data = {
        difficultyMode = 0,
        fromCutscene = false,
        hasFiveHeartGems = false
    }
}

elsTerminaFinalBoss.fieldInformation = {
    difficultyMode = {
        fieldType = "integer",
        options = {0, 1, 2},
        fieldOptions = {
            [0] = "Normal (Darkness ELS)",
            [1] = "Morpho ELS",
            [2] = "Celestial Morpho ELS"
        },
        editable = true
    },
    fromCutscene = {
        fieldType = "boolean"
    },
    hasFiveHeartGems = {
        fieldType = "boolean"
    }
}

elsTerminaFinalBoss.fieldOrder = {
    "x",
    "y",
    "difficultyMode",
    "fromCutscene",
    "hasFiveHeartGems"
}

function elsTerminaFinalBoss.selection(room, entity)
    return utils.rectangle(entity.x - 50, entity.y - 70, 100, 140)
end

return elsTerminaFinalBoss

File: Ahorn/entities/ELSTerminaFinalBoss.jl
module ELSTerminaFinalBoss

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/ELSTerminaFinalBoss" ELSTerminaFinalBoss(x::Integer, y::Integer, difficultyMode::Integer=0, fromCutscene::Bool=false, hasFiveHeartGems::Bool=false)

const placements = Ahorn.PlacementDict(
    "default" => Ahorn.EntityPlacement(ELSTerminaFinalBoss)
)

function Ahorn.selection(entity::ELSTerminaFinalBoss)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ELSTerminaFinalBoss, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
